drivers/flash/nrf_qspi_nor.c

```diff
@@ -410,6 +433,74 @@ static int qspi_wait_while_writing(const struct device *dev)
 	return (ret < 0) ? ret : 0;
 }
 
+static int qspi_wrsrX(const struct device *dev, uint8_t sr_val, uint8_t sr_num)
+{
+	int ret = 0;
+	uint8_t opcode = SPI_NOR_CMD_WRSR;
+	uint8_t length = 1;
+	uint8_t sr_array[2] = {0};
+
+	if (sr_num > 2 || sr_num == 0) {
+		return -EINVAL;
+	}
+
+	if (sr_num == 1) {
+		sr_array[0] = sr_val;
+		if (INST_0_QER == JESD216_DW15_QER_S2B1v1) {
```

Comment:
@mbolivar-nordic @tejlmand @gmarull @nordic-krch do we need a macro for this, or is this the right way to go about it?

Reply:
> @mbolivar-nordic @tejlmand @gmarull @nordic-krch do we need a macro for this, or is this the right way to go about it?

This seems okay for an explicitly single-instance driver.